step1/src/main.js

```diff
@@ -1 +1,146 @@
-// 코드 작성
+/** !! 동적으로 생성된 태그에 이벤트 부여가 한 번 밖에 되지 않고 있음 !! **/
+
+
+/** DOM 객체 **/
+var $appender = document.querySelector('form[name="appenderForm"]');
+//querySelectorAll로 받았을 경우 forEach를 통해 각각 이벤트 리스너 부여해야함!
+var $itemList = document.querySelector('#itemList'); //ul
+var $removeBtns = $itemList.querySelectorAll('.remove');
+var $editBtns = $itemList.querySelectorAll('.edit');
+var $cancelBtns = $itemList.querySelectorAll('.cancel');
+var $completeBtns = $itemList.querySelectorAll('.complete');
+var $doneBtns = $itemList.querySelectorAll('.done');
+var $revokeBtn = $itemList.querySelector('.revoke');
+
+/** 이벤트 리스너**/
+$appender.addEventListener('submit', send);
+$removeBtns.forEach(function ($removeBtn) {
+        $removeBtn.addEventListener('click', remove);
+    }
+);
+$editBtns.forEach(function ($editBtn) {
+    $editBtn.addEventListener('click', edit);
+});
+$cancelBtns.forEach(function ($cancelBtn) {
+    $cancelBtn.addEventListener('click', cancel);
+});
+$completeBtns.forEach(function ($completeBtn) {
+    $completeBtn.addEventListener('click', complete);
+});
+$doneBtns.forEach(function ($doneBtn){
+    $doneBtn.addEventListener('click', done);
+});
+//이벤트 추가 함수
+// function register($parent) { // 한 번 밖에 작동 안됨!!!
+//     $parent.querySelector('.complete').addEventListener('click', complete);
+    
+//     $parent.querySelector('.cancel').addEventListener('click', cancel);
+
+//     $parent.querySelector('.done').addEventListener('click', done);
+   
+//     $parent.querySelector('.edit').addEventListener('click', edit);
+
+//     $parent.querySelector('.remove').addEventListener('click', remove);
+
+// }
+
+/** 이벤트 함수 **/
+
+
+function done(e){
+    e.preventDefault();
+    $parent = e.target.parentNode.parentNode.parentNode;
+    var txt = $parent.querySelector('input').value;
+    var content =
+        '<p style = "#000000">' + txt +'</p>' +
+        '<button class="complete" type="button">완료</bytton>' +
+        '<button class="edit" type="button">수정</bytton>' +
+        '<button class="remove" type="button">삭제</bytton>';
```

Comment:
```suggestion
    const content = `
        <p style = "#000000">${txt}</p>
        <button class="complete" type="button">완료</bytton>
        <button class="edit" type="button">수정</bytton>
        <button class="remove" type="button">삭제</bytton
    `;
```
template literal을 사용하면
이렇게 표현할 수 있답니다!